velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -362,29 +378,45 @@ struct GetTimestampFunction {
       sessionTimeZone_ = tz::locateZone(sessionTimezoneName);
     }
     if (format != nullptr) {
-      formatter_ = detail::getDateTimeFormatter(
-                       std::string_view(*format),
-                       legacyFormatter_ ? DateTimeFormatterType::STRICT_SIMPLE
-                                        : DateTimeFormatterType::JODA)
-                       .thenOrThrow(folly::identity, [&](const Status& status) {
-                         VELOX_USER_FAIL("{}", status.message());
-                       });
-      isConstantTimeFormat_ = true;
+      auto formatter = detail::getDateTimeFormatter(
+          std::string_view(*format),
+          legacyFormatter_ ? DateTimeFormatterType::STRICT_SIMPLE
+                           : DateTimeFormatterType::JODA);
+      // Legacy formatter returns null for invalid format but Joda formatter
+      // throws user error.
+      if (formatter.hasError()) {
+        if (legacyFormatter_) {
+          invalidFormat_ = true;
+        } else {
+          VELOX_USER_FAIL(formatter.error().message());
+        }
```

Comment:
Looks like above code could be extracted as a reusable function for L407 - L417.

Reply:
Thanks. Refactored this part.